ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -53,23 +92,132 @@ export class ImportedDocuments extends React.Component {
     if (!importCompleted) {
       const nextEntry = _get(data, 'loaded_entries', 0);
       const response = await importerApi.check(taskId, nextEntry);
+      const responseData = response.data;
+
+      if (responseData) {
+        responseData.records = _get(responseData, 'records', []);
+      }
       if (response.data.status !== 'RUNNING') {
         this.setState({
           importCompleted: true,
           isLoading: false,
-          data: response.data,
+          data: responseData,
         });
       } else {
         this.setState({
-          data: response.data,
           isLoading: true,
+          data: responseData,
         });
       }
+      this.modeLabel();
+      this.statusLabel();
+      this.calculateStatistics();
     } else {
       this.intervalId && clearInterval(this.intervalId);
     }
   };
 
+  modeLabel = () => {
+    let { data, mode } = this.state;
+    switch (data.mode) {
+      case 'IMPORT':
+        mode = (
+          <Label color="blue" basic>
+            Import
+          </Label>
+        );
+        break;
+      case 'DELETE':
+        mode = (
+          <Label color="red" basic>
+            Delete
+          </Label>
+        );
+        break;
+      case 'PREVIEW_IMPORT':
+        mode = (
+          <Label color="teal" basic>
+            Preview (import)
+          </Label>
+        );
+        break;
+      case 'PREVIEW_DELETE':
+        mode = (
+          <Label color="teal" basic>
+            Preview (delete)
+          </Label>
+        );
+        break;
+      default:
+        mode = (
+          <Label color="black" basic>
+            Invalid mode
```

Comment:
We do need a default, but it is probably not very nice to show "Invalid mode". Often, in the code, we raise an exception is such cases. Try to check other `default:` in other `switch` cases.